pkg/api/handlers/libpod/images.go

```diff
@@ -713,18 +713,22 @@ func ImageScp(w http.ResponseWriter, r *http.Request) {
 
 	sourceArg := utils.GetName(r)
 
-	rep, source, dest, _, err := domainUtils.ExecuteTransfer(sourceArg, query.Destination, []string{}, query.Quiet, ssh.GolangMode)
+	opts := entities.ScpExecuteTransferOptions{}
+	opts.ParentFlags = []string{}
```

Comment:
Feels more natural to have `ExecuteTransfer` be tolerant of a nil here so callers don't have to explicitly set this

Reply:
I guess it should already tolerate nil, since the only place `ParentFlags` is really used is [here](https://github.com/containers/podman/blob/e0e502be206217a8d67cf9ee2f7fa9e639d58010/pkg/domain/utils/scp.go#L341) 

Which should behave the same if it's nil vs an empty array.

I'll amend to remove the explicit setting you reference though since it should be fine without. Though there are no tests in place to confirm this, the tests for scp in general are pretty sparse.

Reply:
The parent flags that are passed to `ExecuteTransfer` end up getting passed directly to `Transfer` for local transfers though, so I'll have to verify that would tolerate nil as well.

Reply:
It looks like it should be fine because `parentFlags` gets passed to `transferRootless`, which creates new arrays from them [here](https://github.com/containers/podman/blob/054b16e68eae58c207688d641f578d31e3887373/pkg/domain/infra/abi/images.go#L805)

Then it's also used in `transferRootful` to similar effect [here](https://github.com/containers/podman/blob/054b16e68eae58c207688d641f578d31e3887373/pkg/domain/infra/abi/images.go#L845)

Reply:
The signatures don't seem all that bad, but they're certainly not clean either. Wouldn't call it a requirement but wouldn't say no either.